packages/web-components/src/discount-card/index.ts

```diff
@@ -0,0 +1,17 @@
+import { isSsr } from "../utils/index.ts";
+import { DiscountCard } from "./discount-card.ts";
+
+export { Slots } from "./constants.ts";
+export { DiscountCard };
+
+export const register = () => {
+  if (isSsr()) return;
+
+  customElements.define("tapsi-discount-card", DiscountCard);
+};
```

Comment:
We can check if the component is already defined or not.

```suggestion
export const register = () => {
  if (isSsr()) return;
  if (customElements.get("tapsi-discount-card")) return;

  customElements.define("tapsi-discount-card", DiscountCard);
};
```